File: src/CacheStrategies/LeaderBoard.WriteThrough/Dtos/PlayerScoreDto.cs
namespace LeaderBoard.WriteThrough.Dtos;

public record PlayerScoreDto(
    string PlayerId,
    double Score,
    string LeaderBoardName,
    long? Rank = 1,
    string? Country = null,
    string? FirstName = null,
    string? LastName = null
);

File: src/Client/src/app/core/dtos/player-score-dto.ts
export interface PlayerScoreDto {
  playerId: string;
  score: number;
  leaderBoardName: string;
  rank?: number | null;
  firstName: string;
  lastName: string;
  country: string;
}

export interface PlayerScoreWithNeighborsDto {
  previous?: PlayerScoreDto | null;
  currentPlayerScore: PlayerScoreDto;
  next?: PlayerScoreDto | null;
}
